### PR TITLE
add test coverage table to github summary

### PR DESCRIPTION
**Pull request by derekk-nm**
This PR introduces the creation of a JSON form of the test coverage information that has only been generated as HTML up until this point.  The JSON content is then parsed by a script to generate a table that is compatible with `github-markdown` so that it can be included in the test job summary.  The table shows the overall test coverage, and the coverage provided by each sub-directory at the level just below vllm.
[This nm remote push](https://github.com/neuralmagic/nm-vllm/actions/runs/9844380211) job demonstrates that the code changes are working (no failures), and shows the expected test coverage summary table (on the summary page, scroll down to the TEST summary), and JSON artifact (on the summary page, scroll down to the artifacts listing and look for the files starting with `cc-vllm-json-*`.

Also included are some additional functions that will be excluded/omitted from the test coverage.  see pyproject.toml

### CODE REVIEW

File: .github/actions/nm-summary-test/action.yml [hunk 1]
```diff
@@ -42,4 +43,14 @@ runs:
...
       echo "| test: | ${TEST_EMOJI} |" >> $GITHUB_STEP_SUMMARY
+      echo "" >> $GITHUB_STEP_SUMMARY
+      echo "#### TestMo URL: ${TESTMO_URL}" >> $GITHUB_STEP_SUMMARY
+      echo "## Test Coverage" >> $GITHUB_STEP_SUMMARY
+      # coverage_report_breakdown.py requires the `tabulate` package
+      # to generate the markdown for the summary.
+      pip3 install tabulate
+      # As a multiline response we cannot pass the table directly to github
+      # so redirect it to a file, then cat the file to the output
+      python3 ./.github/scripts/coverage_report_breakdown.py ${{ inputs.coverage_json }} > COVERAGE_MD
```
Comment (andy-neuma) [inline .github/actions/nm-summary-test/action.yml:54]: the table ends up looking nice.

File: .github/scripts/coverage_report_breakdown.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("coverage_json_file",
+                        type=str,
+                        help="file path to coverage JSON output")
+    args = parser.parse_args()
+    cc = CodeCoverage(Path(args.coverage_json_file))
+
+    print(cc.to_github_markdown())
```
Comment (andy-neuma) [inline .github/scripts/coverage_report_breakdown.py:174]: i like that you are outputting to a file. this makes it easy to run locally.

File: .github/scripts/coverage_report_breakdown.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        empty_row_df = pd.Series(
+            pd.NA, index=summary_df.columns).to_frame().transpose()
+        header_row_df = empty_row_df.copy()
+        header_row_df["Collection"] = "vllm Subdirs"
+        summary_df = pd.concat(
+            [summary_df, empty_row_df, header_row_df, breakdown_df],
+            ignore_index=True)
+        # clean up the `nan` values for display purposes
+        summary_df = summary_df.astype(str)
+        summary_df.replace({"nan": None}, inplace=True)
```
Comment (andy-neuma) [inline .github/scripts/coverage_report_breakdown.py:155]: quick question, how do we generate "nan"?

**Review by andy-neuma — COMMENTED** [review]
looking good

File: .github/scripts/coverage_report_breakdown.py (hunk 3 above)
Comment (derekk-nm) [reply .github/scripts/coverage_report_breakdown.py:155]: this was "fun" 😉 .  to get that empty row and header row between the overall info and the per-sub-directory breakdown I needed to add that `empty_row_df` and `header_row_df`.  To get the `concat` to work successfully I needed to populate those with `pd.NA`, (which is `nan`).  once the `concat` was done, I could clean up those `nan`s

**Review by andy-neuma — APPROVED** [review]
looks good to me.

**Review by dbarbuzzi — APPROVED** [review]

**Comment by derekk-nm** [comment]
moved to nm-vllm-ent

**Pull request closed without merging**